src/librustc_middle/ty/relate.rs

```diff
@@ -559,7 +560,27 @@ pub fn super_relate_consts<R: TypeRelation<'tcx>>(
                     }
                 }
 
-                // FIXME(const_generics): handle `ConstValue::ByRef`.
+                (ConstValue::ByRef { .. }, ConstValue::ByRef { .. }) => {
+                    // `a` and `b` are already fully resolved, so we can
+                    // use `reveal_all` here.
+                    let param_env = ty::ParamEnv::reveal_all();
```

Comment:
You need to check `if !(a.ty, b.ty).has_param_types_or_consts()`, otherwise the layout could depend on e.g. a `*mut T` layout where `T: SIzed` is in the `ParamEnv`.

Oh but you have a `param_env` here. So why not use that?

Reply:
isn't this impossible? Afaik const args cannot have a generic type

Reply:
I also don't know how to use `ParamEnv::new` as I am missing a useable `DefId`